### ISSUE TITLE
some problem about yolo-nas training

### ISSUE THREAD

**Issue by xywlpo** [Deci-AI/super-gradients#1481]
### 💡 Your Question

Thanks for your great work, the model seems very powerful. I want to use your pretrain model to train my own datasets with yolo-nas-m. But the result of map0.5-0.95 is lower than ppyoloe-plus-m about 6%. Could you give me some advice about training parameters? My training code is below.

`    

    init_trainer()

    # Launch DDP on 4 GPUs'
    setup_device(device='cuda', multi_gpu=MultiGPUMode.DISTRIBUTED_DATA_PARALLEL, num_gpus=4)
    
    # create trainer
    trainer = Trainer(experiment_name='yolo-nas-m', ckpt_root_dir='train_log_20230919')
    
    class_inclusion_list = ["yugan"]
    # create train dataloader
    train_dataset_params = {
        'data_dir': 'fishing_dataset',
        'subdir': 'train/images',
        'json_file': 'train.json', 
        'input_dim': [736, 1280],
        "class_inclusion_list": class_inclusion_list,
        'ignore_empty_annotations': False
    }
    train_data = coco2017_train_yolo_nas(
        dataset_params=train_dataset_params,
        dataloader_params={'batch_size':20}
    )
    
    # create val dataloader
    val_dataset_params = {
        'data_dir': 'fishing_dataset',
        'subdir': 'val/images',
        'json_file': 'val.json', 
        'input_dim': [736, 1280],
        "class_inclusion_list": class_inclusion_list,
        'ignore_empty_annotations': False
    }    
    val_data = coco2017_val_yolo_nas(
        dataset_params=val_dataset_params,
        dataloader_params={'batch_size':10}
    )

    # create model
    model = models.get('yolo_nas_m', num_classes=1, pretrained_weights="coco")

    # train parameters
    train_params = {
        'silent_mode': False,
        "max_epochs": 300,
        
        # warmup parameters
        "warmup_mode": "linear_epoch_step",
        "warmup_initial_lr": 1e-6,
        "lr_warmup_epochs": 1,
        
        # learning rate
        "initial_lr": 5e-4,
        "lr_mode": "cosine",
        "cosine_final_lr_ratio": 0.1,
        "zero_weight_decay_on_bias_and_bn": True,
        "batch_accumulate": 1,
        
        # loss setting
        "loss": PPYoloELoss(
            use_static_assigner=False,
            num_classes=1,
            reg_max=16
        ),    
        
        # optimizer
        "optimizer": "Adam",
        "optimizer_params": {"weight_decay": 0.0001},
        
        # ema parameters
        "ema": True,
        "ema_params": {"decay": 0.9997, "decay_type": "threshold"},
        "zero_weight_decay_on_bias_and_bn": True,
        "average_best_models":True,
        
        # mixed precision
        "mixed_precision": True,
        "sync_bn": True,
        
        "run_validation_freq": 1,
        
        # valid metrics
        "valid_metrics_list": [
            DetectionMetrics(
                score_thres=0.1,
                top_k_predictions=300,
                num_cls=1,
                normalize_targets=True,
                include_classwise_ap=True,
                post_prediction_callback=PPYoloEPostPredictionCallback(
                    score_threshold=0.01,
                    nms_top_k=1000,
                    max_predictions=300,
                    nms_threshold=0.7
                )
            )
        ],
        "metric_to_watch": "mAP@0.50:0.95",
        "greater_metric_to_watch_is_better": True
    }

    # start to train
    trainer.train(model=model, 
                  training_params=train_params, 
                  train_loader=train_data, 
                  valid_loader=val_data)
  `

**Comment by BloodAxe** [comment]
I'm afraid there is not enough information to suggest a right direction at this point. Can you attach a metrics plot and some insights about your dataset (You can use DataGradients to generate a report once you instantiated train_data  / val_data btw).

So a few ideas what it could be:

* Maybe there is an overfitting? Try playing with transforms you are using during training. Maybe default transforms are not optimal for your dataset. Try with and without mosaic and mixup. Maybe scale range in RandomAffine is too broad for your taks.

* Maybe you are lowering LR too quickly? Try different starting/final learning rates.

* Maybe EMA parameter is too low/high? Try playing changing that `decay` value for EMA and see how it impacts the metric

* You can also employ [ExtremeBatchDetectionVisualizationCallback](https://github.com/Deci-AI/super-gradients/blob/master/src/super_gradients/training/utils/callbacks/callbacks.py#L1198) to see what is happening with predictions during training. This helps to see how samples with the worst loss or metric score looks like.

**Comment by BloodAxe** [comment]
Closing as stale
